main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -172,20 +177,48 @@ object TypeReduction {
               case (clazz, tpe) => Type.getFlixType(clazz) == tpe
             })
         exactMethod.length match {
-          case 1 => JavaResolutionResult.Resolved(Type.Cst(TypeConstructor.JvmMethod(exactMethod.head), loc))
-          case _ => JavaResolutionResult.AmbiguousMethod(candidateMethods.toList) // 0 corresponds to no exact method, 2 or higher should be impossible in Java
+          case 1 => JavaMethodResolutionResult.Resolved(Type.Cst(TypeConstructor.JvmMethod(exactMethod.head), loc))
+          case _ => JavaMethodResolutionResult.AmbiguousMethod(candidateMethods.toList) // 0 corresponds to no exact method, 2 or higher should be impossible in Java
         }
     }
   }
 
-  /**
+    /**
+     * Helper method to retrieve a constructor given its parameter types and the class.
+     * Returns a JavaConstructorResolutionResult containing either the constructor, a list of candidate constructors or
+     * a ConstructorNotFound object. The working process is similar to retrieveMethod and differs in the selection of candidate constructors.
+     */
+  private def retrieveConstructor(clazz: Class[_], ts: List[Type], loc: SourceLocation)(implicit flix: Flix): JavaConstructorResolutionResult = {
```

Comment:
I wonder if we can do some refactor to better share code. What do you think?

Reply:
Hm, I also thought about later refactoring.
We may do it if we use match patterns, however method resolution involves a method name which differs from constructor resolution.
We may also consider a JavaResolutionProcess with an option type which could hold either a constructor or a method, depending on the called function the context would de-ambiguize it.
The central point for factorization would be isCandidateMethod/isCandidateConstructor holding different conditions, if we can use also a match pattern here then we could factorize it with a generic JavaResolutionProcess.